recipes/imagl/all/conanfile.py

```diff
@@ -117,10 +133,31 @@ def requirements(self):
     def _configure_cmake(self):
         if self._cmake:
             return self._cmake
+
+        # CMake generator must be set to Ninja when using Visual Studio to let choose the right 
+        # MSVC compiler version when multiple versions are installed on build machine. The problem
+        # with default generator is that it will always choose the last MSVC version. It will 
+        # generate Visual Studio solution and project files, and use MSBuild to build it. I think 
+        # it's a "feature" of CMake / Visual Studio generator that does not use environment which 
+        # is set by vcvars. So if you want to use a specific MSVC version, you have to use Ninja 
+        # generator with CMake. You can find some side explanation here: 
+        # https://devblogs.microsoft.com/cppblog/side-by-side-minor-version-msvc-toolsets-in-visual-studio-2019/
+        # 
+        # Building with specific MSVC version could be needed since imaGL requires some minimal 
+        # version to be compiled according to its own version. Following table links imaGL version 
```

Comment:
Is this required for the build to pass on CCI?

In the Conan universe if you want to build an exact toolset you could use the new msvc compiler. (this is consistent with CMake)

Just as a sanity check, 

Is this required ? Or is it an improvement?

Reply:
No, it's not required to pass on CCI. By the way, CCI is not able to build imaGL since `tools.vswhere` does not work on CCI (see #4954) and I don't know how to test for minor version of MSVC without vswhere. It's done from line [89](https://github.com/conan-io/conan-center-index/blob/0dcdf201d5fa73f758a7cdcc7af48cfc573e13b3/recipes/imagl/all/conanfile.py#L89) to [95](https://github.com/conan-io/conan-center-index/blob/0dcdf201d5fa73f758a7cdcc7af48cfc573e13b3/recipes/imagl/all/conanfile.py#L95). Some imaGL versions could be built in CCI (CCI has Visual Studio 16.4, so imaGL 0.1.2 and 0.2.1 could be built), but I don't know how to be sure that the recipe is cooked on the CCI CI/CD machine.

I do this in the recipe to allow imaGL to be built locally in a Visual Studio command line environment with a specific version of the toolset (using VCVARS and company environment variables). So the recipe is compatible with "Visual Studio" compiler from the Conan universe.

Still in this universe, using the new msvc compiler is allowed thanks to lines [47](https://github.com/conan-io/conan-center-index/blob/0dcdf201d5fa73f758a7cdcc7af48cfc573e13b3/recipes/imagl/all/conanfile.py#L47) and [53](https://github.com/conan-io/conan-center-index/blob/0dcdf201d5fa73f758a7cdcc7af48cfc573e13b3/recipes/imagl/all/conanfile.py#L53). But CCI still uses "Visual Studio".